lib/spack/spack/util/web.py

```diff
@@ -235,7 +234,7 @@ def push_to_url(local_file_path, remote_path, keep_original=True, extra_args=Non
     remote_url = urllib.parse.urlparse(remote_path)
     if remote_url.scheme == "file":
         remote_file_path = url_util.local_file_path(remote_url)
-        mkdirp(os.path.dirname(remote_file_path))
+        mkdirp(PurePath(remote_file_path).parent)
```

Comment:
Cast the method invocation here as well. 
Additionally, can we use pathlib's mkdir here? Spack has a special method because `os.mkdir` lacked functionality, but if pathlib vendors that functionality, we should use it with the goal of eventually deprecating the parts of `llnl.util` we don't need.

Reply:
I tried to switching to Pathlib's mkdir but ended up getting errors in the CI check: prechecks / bootstrap-dev-rhel8. It seems to me that the pathlib version just mimics os.mkdir